process/elasticproc/block/blockProcessor.go

```diff
@@ -227,17 +230,53 @@ func (bp *blockProcessor) getEncodedMBSHashes(body *block.Body) []string {
 	return miniblocksHashes
 }
 
-func putMiniblocksDetailsInBlock(header coreData.HeaderHandler, block *data.Block) {
+func putMiniblocksDetailsInBlock(header coreData.HeaderHandler, block *data.Block, pool *outport.Pool, body *block.Body) {
 	mbHeaders := header.GetMiniBlockHeaderHandlers()
+
 	for idx, mbHeader := range mbHeaders {
+		mbType := nodeBlock.Type(mbHeader.GetTypeInt32())
+		if mbType == nodeBlock.PeerBlock {
+			continue
+		}
+
+		txsHashes := body.MiniBlocks[idx].TxHashes
 		block.MiniBlocksDetails = append(block.MiniBlocksDetails, &data.MiniBlocksDetails{
-			IndexFirstProcessedTx: mbHeader.GetIndexOfFirstTxProcessed(),
-			IndexLastProcessedTx:  mbHeader.GetIndexOfLastTxProcessed(),
-			MBIndex:               idx,
+			IndexFirstProcessedTx:    mbHeader.GetIndexOfFirstTxProcessed(),
+			IndexLastProcessedTx:     mbHeader.GetIndexOfLastTxProcessed(),
+			MBIndex:                  idx,
+			ProcessingType:           nodeBlock.ProcessingType(mbHeader.GetProcessingType()).String(),
+			Type:                     mbType.String(),
+			SenderShardID:            mbHeader.GetSenderShardID(),
+			ReceiverShardID:          mbHeader.GetReceiverShardID(),
+			TxsHashes:                hexEncodeSlice(txsHashes),
+			ExecutionOrderTxsIndices: extractExecutionOrderIndicesFromPool(mbHeader, txsHashes, pool),
 		})
 	}
 }
 
+func extractExecutionOrderIndicesFromPool(mbHeader coreData.MiniBlockHeaderHandler, txsHashes [][]byte, pool *outport.Pool) []int {
+	txsMap := getTxsMap(nodeBlock.Type(mbHeader.GetTypeInt32()), pool)
+	executionOrderTxsIndices := make([]int, len(txsHashes))
+	indexOfFirstProcessed, indexOfLastProcessed := mbHeader.GetIndexOfFirstTxProcessed(), mbHeader.GetIndexOfLastTxProcessed()
+	for idx, txHash := range txsHashes {
+		tx, found := txsMap[string(txHash)]
+		if !found {
+			log.Warn("blockProcessor.extractExecutionOrderIndicesFromPool cannot find tx in pool", "txHash", hex.EncodeToString(txHash))
```

Comment:
for the not found Tx it will appear as the order is 0, can we mark it as not found?
e.g create a const txNotFound = -2 ?

Reply:
added